src/controllers/video.controller.js

```diff
@@ -1,40 +1,317 @@
+    const thumbnailFile = await uploadOnCloudinary(thumbnailLocalPath)
+
+    if (!videoFile) {
+        throw new ApiError(400, "Error while uploading Video on cloudinary")
+    }
+
+    if (!thumbnailFile) {
+        throw new ApiError(400, "Error while uploading Thumbnail on cloudinary")
+    }
+
+    //create object and send on video collection DB
+    const video = await Video.create({
+        video: videoFile.url,
+        thumbnail: thumbnailFile.url,
+        publicId: videoFile.public_id,
+        title,
+        description,
+        duration: videoFile.duration,
+        owner: req.user?._id,
+    })
+
+    const videoUploaded = await Video.findById(video?._id).select("-video -thumbnail -views -isPublished")
+
+    if (!videoUploaded) {
+        throw new ApiError(500, "Video is not Uploaded !")
+    }
+
+    return res
+        .status(200)
+        .json(
+            new ApiResponse(200, videoUploaded, "Video Uploaded sucessfully !")
+        )
+
+})
+
 const getAllVideos = asyncHandler(async (req, res) => {
-    const { page = 1, limit = 10, query, sortBy, sortType, userId } = req.query
+    const { page = 1, limit = 10, sortBy = "title", sortType = "ascending", userId } = req.query
     //TODO: get all videos based on query, sort, pagination
-})
 
-const publishAVideo = asyncHandler(async (req, res) => {
-    const { title, description} = req.body
-    // TODO: get video, upload to cloudinary, create video
+    const pageNumber = parseInt(page)
+    const pageLimit = parseInt(limit)
+    const skip = (pageNumber - 1) * pageLimit
+    const sortdirection = sortType === "ascending" ? 1 : -1
+
+    if (!isValidObjectId(userId)) {
+        throw new ApiError(400, "usedId is not found, userId is required !")
+    }
+
+    try {
+        const videos = await Video.aggregate(
+            [
+                {
+                    $match: {
+                        owner: new mongoose.Types.ObjectId(userId)
+                    }
+                },
+                {
+                    $lookup: {
+                        from: "users",
+                        localField: "owner",
+                        foreignField: "_id",
+                        as: "owner",
+                        pipeline: [
+                            {
+                                $project: {
+                                    fullname: 1,
+                                    username: 1,
+                                    avatar: 1
+                                }
+                            }
+                        ]
+                    }
+                },
+                {
+                    $skip: skip
+                },
+                {
+                    $limit: pageLimit
+                },
+                {
+                    $sort: { [sortBy]: sortdirection }
+                }
+
+            ])
+
+        const totalVideos = await Video.countDocuments({ owner: userId })
```

Comment:
shouldn't be it videos.length, which you have fetched
why the total videos from video model?